### PR TITLE
Added Len for counting the length of the byte array

### PR DESCRIPTION
**Pull request by gertjanal**
Couldn't find any byte length expression, let me know if I was right so I can create tests

### CODE REVIEW

File: src/main/java/io/parsingdata/metal/expression/value/reference/Len.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private final String _name;
+
+    public Len(final String name) {
+        _name = checkNotNull(name, "name");
+    }
+
+    @Override
+    public OptionalValue eval(final Environment env, final Encoding enc) {
+        final ParseValue value = env.order.get(_name);
+        return OptionalValue.of(value == null ? null : ConstantFactory.createFromNumeric(value.getValue().length, new Encoding(true)));
```
Comment (gertjanal) [inline src/main/java/io/parsingdata/metal/expression/value/reference/Len.java]: null check

File: src/main/java/io/parsingdata/metal/expression/value/reference/Len.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import io.parsingdata.metal.data.Environment;
+import io.parsingdata.metal.data.ParseValue;
+import io.parsingdata.metal.encoding.Encoding;
+import io.parsingdata.metal.expression.value.ConstantFactory;
+import io.parsingdata.metal.expression.value.OptionalValue;
+import io.parsingdata.metal.expression.value.Value;
+import io.parsingdata.metal.expression.value.ValueExpression;
+
+public class Len implements ValueExpression {
```
Comment (jvdb) [inline src/main/java/io/parsingdata/metal/expression/value/reference/Len.java]: This should probably extend `UnaryValueExpression`.

Comment (gertjanal) [inline src/main/java/io/parsingdata/metal/expression/value/reference/Len.java]: updated

**Comment by jvdb** [comment]
Implements part of #37.

**Pull request merged**